internal/state/state.go

```diff
@@ -88,8 +90,21 @@ func (c *state) newClient() (hcapi2.Client, error) {
 		if filePath == "" {
 			opts = append(opts, hcloud.WithDebugWriter(os.Stderr))
 		} else {
-			writer, _ := os.Create(filePath)
-			opts = append(opts, hcloud.WithDebugWriter(writer))
+			f, err := os.OpenFile(filePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+			if err != nil {
+				return nil, err
+			}
+
+			quotedArgs := make([]string, 0, len(os.Args))
+			for _, arg := range os.Args {
+				quotedArgs = append(quotedArgs, fmt.Sprintf("%q", arg))
+			}
+			_, err = f.WriteString("--- Command:\n" + strings.Join(quotedArgs, " ") + "\n\n\n\n")
+			if err != nil {
+				return nil, err
+			}
```

Comment:
This only happens for `HCLOUD_DEBUG_FILE`, not for `HCLOUD_DEBUG`. We might need to break up the condition.